### PR TITLE
Add section prototype

### PR DESCRIPTION
**Pull request by haipeng-wang**

### CODE REVIEW

File: ui/src/component/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Pie from './pie';
+
+const chartMapper = {
+  pie: props => (<Pie {...props} />),
+};
```
Comment (lizho-ling) [inline ui/src/component/index.js]:
```
const chart2Component = {
   pie: 'Pie',
}

render() {
    const ChartComponent = _.result(chart2Component, chartType, 'line');
    return (<div>{  <ChartComponent {...props} }</div>);
}
```
Move chart initialize to render function, if we need to pass more parameters to chart,we only need modifying one place

File: ui/src/section/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    this.onSlicerChange = this.onSlicerChange.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!_.isEqual(nextProps.config, this.props.config)) {
+      this.updateMetrics(nextProps.config);
+    }
+  }
+
+  onSlicerChange(args) {
```
Comment (lizho-ling) [inline ui/src/section/index.js]: ```onSlicerChange = (args) => {}```
Then you don't need line 19

File: ui/src/section/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    // TODO: just placeholder here
+    const { name, slicer } = args;
+    this.updateMetrics(_.defaults({
+      selected: name,
+    }, this.props.config));
+    const { history } = this.state;
+    history.push({ name, slicer });
+    this.setState({
+      history,
+    });
```
Comment (lizho-ling) [inline ui/src/section/index.js]:
```
this.setState({
   history: [...history, {name, slice}],
})
```
Set new state every time

File: ui/src/section/index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    config: PropTypes.objectOf(PropTypes.object).isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      source: [],
+      history: [],
+    };
+    this.updateMetrics(props.config);
```
Comment (lizho-ling) [inline ui/src/section/index.js]: Do this in componentDidMount

File: ui/stories/section.stories.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { storiesOf } from '../.storybook/facade';
+import Section from '../src/section';
+
+
+storiesOf('Section', module)
+  .add('pie chart', () => (
+    <Section config={{
+      section: {
+        dimensions: {
```
Comment (lizho-ling) [inline ui/stories/section.stories.js]: Apart from section key, will we have any other top level keys?
If not, pls move dimensions and chartType to top level

Comment (haipeng-wang) [reply ui/stories/section.stories.js]: We will have some other props from story level, it will update when we finalize story define

**Pull request merged**